### PR TITLE
[Transform] Make validate calls cancellable

### PR DESCRIPTION
**Pull request by prwhelan**
Validate will initiate a search request.  In the event that the search request needs to be cancelled, rather than manually stopping the task, cancelling the Validate task will now propagate the cancel request to the Search task.

Relate #88010

### CODE REVIEW

File: docs/changelog/110951.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 110951
+summary: Make validate calls cancellable
```
Comment (jonathan-buttner) [inline docs/changelog/110951.yaml:2]: Maybe: `Allow task canceling of validate API calls`?

**Review by jonathan-buttner — APPROVED** [review]
Looks good, any easy tests we could add to ensure we don't accidentally start passing around the wrong task id?

**Comment by prwhelan** [comment]
@elasticmachine update branch

**Comment by prwhelan** [comment]
> Looks good, any easy tests we could add to ensure we don't accidentally start passing around the wrong task id?

We spoke offline - there's no good way to test this, and the other PRs on https://github.com/elastic/elasticsearch/issues/88010 do not have any examples either :(

**Review by maxhniebergall — APPROVED** [review]
LGTM

**Pull request merged**